cmd/server/run.go

```diff
@@ -153,19 +151,27 @@ func ProcessQueue(queue <-chan func() error) {
 }
 
 // PublishIssues publishes the issues in the PR for the given project branch
-func PublishIssues(ctx context.Context, sonar *sonarqube2.Sonarqube, projectScm scm2.SCM, project string, branch string) error {
+func PublishIssues(ctx context.Context, sonar *sonarqube2.Sonarqube, projectScm scm2.SCM, project string, branch string, branchType string) error {
 	// Find PR
-	pr, err := sonar.FindPRForBranch(project, branch)
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to find PR for branch %s of the project %s", branch, project))
+	var pr *sonarqube2.PullRequest
+	var err error
+	if branchType == sonarqube2.BRANCH_TYPE_PULL_REQUEST {
+		pr, err = sonar.FindPRForKey(project, branch)
```

Comment:
second parameter name can create confusion, but it really came in under 'branch' json key

Reply:
Maybe it would be better to change the signature of `PublishIssues` function and instead of `branch` and `branchType` strings just pass `Branch` struct with data from webhook.